op-node/rollup/derive/channel_test.go

```diff
@@ -99,3 +106,60 @@ func TestFrameValidity(t *testing.T) {
 		t.Run(tc.name, tc.Run)
 	}
 }
+
+func TestBatchReader(t *testing.T) {
+	// Get batch data
+	rng := rand.New(rand.NewSource(0x543331))
+	singularBatch := RandomSingularBatch(rng, 5000, big.NewInt(333))
+	batchDataInput := NewBatchData(singularBatch)
+
+	encodedBatch := &bytes.Buffer{}
+	// Get the encoded data of the batch data
+	batchDataInput.encodeTyped(encodedBatch)
+
+	var testCases = []struct {
+		name string
+		algo func(buf *bytes.Buffer)
```

Comment:
If you also pass a `t *testing.T` to the func, you can assert in the funcs that the write operations don't error. Or alternatively, return `error` and assert in the main test.